moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -91,10 +91,12 @@ double RobotTrajectory::getDuration() const
 
 double RobotTrajectory::getAverageSegmentDuration() const
 {
-  if (duration_from_previous_.empty())
+  if (duration_from_previous_.size() <= 1)
+  {
```

Comment:
Sorry, but I just found this option for logging with no Node:  https://answers.ros.org/question/355985/global-logger-for-logging-without-a-node/?answer=355986#post-id-355986

```suggestion
  {
    RCLCPP_WARN(rclcpp::get_logger("my_logger"), "Too few waypoints to calculate a duration. Returning 0.");
```